scene/gui/progress_bar.cpp

```diff
@@ -60,15 +60,55 @@ void ProgressBar::_notification(int p_what) {
 		Color font_color = get_theme_color("font_color");
 
 		draw_style_box(bg, Rect2(Point2(), get_size()));
+
 		float r = get_as_ratio();
-		int mp = fg->get_minimum_size().width;
-		int p = r * (get_size().width - mp);
-		if (p > 0) {
-			if (is_layout_rtl()) {
-				draw_style_box(fg, Rect2(Point2(p, 0), Size2(fg->get_minimum_size().width, get_size().height)));
-			} else {
-				draw_style_box(fg, Rect2(Point2(0, 0), Size2(p + fg->get_minimum_size().width, get_size().height)));
-			}
+
+		switch (mode) {
+			case FILL_LEFT_TO_RIGHT: {
+				int mp = fg->get_minimum_size().width;
+				int p = round(r * (get_size().width - mp));
+				int p_remaining = round((1.0 - r) * (get_size().width - mp));
+
+				if (p > 0) {
+					if (is_layout_rtl()) {
+						draw_style_box(fg, Rect2(Point2(p_remaining, 0), Size2(p + fg->get_minimum_size().width, get_size().height)));
+					} else {
+						draw_style_box(fg, Rect2(Point2(0, 0), Size2(p + fg->get_minimum_size().width, get_size().height)));
+					}
+				}
+			} break;
+			case FILL_RIGHT_TO_LEFT: {
+				int mp = fg->get_minimum_size().width;
+				int p = round(r * (get_size().width - mp));
+				int p_remaining = round((1.0 - r) * (get_size().width - mp));
+
+				if (p > 0) {
+					if (is_layout_rtl()) {
+						draw_style_box(fg, Rect2(Point2(0, 0), Size2(p + fg->get_minimum_size().width, get_size().height)));
+					} else {
+						draw_style_box(fg, Rect2(Point2(p_remaining, 0), Size2(p + fg->get_minimum_size().width, get_size().height)));
+					}
+				}
+			} break;
+			case FILL_TOP_TO_BOTTOM: {
+				int mp = fg->get_minimum_size().height;
+				int p = round(r * (get_size().height - mp));
+
+				if (p > 0) {
+					draw_style_box(fg, Rect2(Point2(0, 0), Size2(get_size().width, p + fg->get_minimum_size().height)));
+				}
+			} break;
+			case FILL_BOTTOM_TO_TOP: {
+				int mp = fg->get_minimum_size().height;
+				int p = round(r * (get_size().height - mp));
+				int p_remaining = round((1.0 - r) * (get_size().height - mp));
+
+				if (p > 0) {
+					draw_style_box(fg, Rect2(Point2(0, p_remaining), Size2(get_size().width, p + fg->get_minimum_size().height)));
+				}
+			} break;
+			case FILL_MODE_MAX:
+				break;
```

Comment:
This doesn't look necessary.

Reply:
If I didn't add this, it couldn't pass the build checks somehow.